examples/java/PluginServer.java

```diff
@@ -0,0 +1,547 @@
+                    } else {
+                        logger.log(Level.WARNING, "The script request header is null.");
+                        responseObserver.onError(new Throwable("The script request header is null in onNext in evaluateScript."));
+                        responseObserver.onCompleted();
+                    }
+                }
+                
+                @Override
+                public void onError(Throwable t) {
+                    logger.log(Level.WARNING, "Encountered error in evaluateScript", t);
+                    responseObserver.onCompleted();
+                }
+                
+                @Override
+                public void onCompleted() {
+                    logger.fine("onCompleted in evaluateScript called");
+                    responseObserver.onCompleted();
+                    logger.fine("onCompleted in evaluateScript completed");
+                }
+            };
+        }
+        
+        private ServerSideExtension.BundledRows evalScript(ServerSideExtension.ScriptRequestHeader header) {
+            
+            logger.fine("evalScript called");
+            ServerSideExtension.BundledRows.Builder builder = ServerSideExtension.BundledRows.newBuilder();
+            ScriptEngineManager manager = new ScriptEngineManager();
+            ScriptEngine engine = manager.getEngineByName("JavaScript");
+            String result;
+            try {
+                Object res = engine.eval(header.getScript());
+                result = res.toString();
+                logger.finer("The string representation of the result: " + result);
+            } catch (Exception e) {
+                logger.log(Level.WARNING, "eval script did not work.", e);
+                return builder.build();
+            }
+            
+            ServerSideExtension.Row.Builder rowBuilder = ServerSideExtension.Row.newBuilder();
+            ServerSideExtension.Dual.Builder dualBuilder = ServerSideExtension.Dual.newBuilder();
+            
+            switch (header.getReturnType()) {
+                case STRING : 
+                    builder.addRows(rowBuilder.addDuals(dualBuilder.setStrData(result)));
+                    break;
+                case NUMERIC : 
+                    builder.addRows(rowBuilder.addDuals(dualBuilder.setNumData(Double.parseDouble(result))));
+                    break;
+                case DUAL : 
+                    builder.addRows(rowBuilder.addDuals(dualBuilder.setStrData(result).setNumData(Double.parseDouble(result))));
+                    break;
+                default :
+                    logger.log(Level.WARNING, "Incorrect return type.");
+                    return builder.build();
+            }
+            logger.fine("evalScript completed");
+            return builder.build();
+        }
+        
+        private ServerSideExtension.BundledRows evalScript(ServerSideExtension.ScriptRequestHeader header, ServerSideExtension.BundledRows bundledRows) {

```

Comment:
Seems like much code in common with previous method. Could it be refactored to avoid code duplication?